### PR TITLE
compiles - but still warnings

### PR DESCRIPTION
**Pull request by D300**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -59,19 +59,19 @@ send me a DM to check your pull request
...
 };
 
-struct <#structName1#>                                //4
+struct Comparator                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
```
Comment (matkatmusic) [inline main.cpp:77]: What do we always do before we use a pointer?
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -81,49 +81,78 @@ struct <#structName1#>                                //4
...
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunctionA(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's nameOne value: " << that->nameOne << std::endl;
+        that->nameOne = *updatedValue;
+        std::cout << "U's nameOne updated value: " << that->nameOne << std::endl;
```
Comment (matkatmusic) [inline main.cpp:122]: What do we always do before we use pointers?
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -81,49 +81,78 @@ struct <#structName1#>                                //4
...
+    T rabbit(2, "Osterhase" );                                             //6
+    T dolphin(10, "Flipper" );                                             //6
+    
+    Comparator f;                                            //7
+    auto* smaller = f.compare(&rabbit, &dolphin);                              //8
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
+    if (smaller != nullptr)
     std::cout << "the smaller one is << " << smaller->name << std::endl; //9
```
Comment (matkatmusic) [inline main.cpp:145]: You added an `if( expr )` but didn't indent the `if` body.
- [ ] fix the formatting.

File: main.cpp [hunk 4]
```diff
@@ -81,49 +81,78 @@ struct <#structName1#>                                //4
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float nameOne { 0 }, nameTwo { 0 };
+    float* memberFunction(float* updatedValue)      //12
+    {
+        return updatedValue;
+    }
```
Comment (matkatmusic) [inline main.cpp:88]: you are not supposed to add a member function to this class.  Just fill in the placeholders.
- [ ] revise

Comment (D300) [reply main.cpp:88]: I thought I didnt add it, I just added a body and return the returntype for printing it out. shouldnt I print it?

File: main.cpp [hunk 5]
```diff
@@ -1,133 +1,172 @@
...
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float nameOne { 0.f }, nameTwo { 0.f };
+    float* memberFunction(float* updatedValue);      //12
```
Comment (matkatmusic) [inline main.cpp:89]: - [ ] Remove this declaration.

File: main.cpp [hunk 6]
```diff
@@ -1,133 +1,172 @@
...
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunctionA(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        
+        if (that != nullptr)
```
Comment (matkatmusic) [inline main.cpp:122]: Did you check `updatedValue` before using it? 
- [ ] revise

File: main.cpp [hunk 7]
```diff
@@ -1,133 +1,172 @@
...
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
+            write something that makes the distance between that->nameTwo and that->nameOne get smaller
              */
-            that-><#name2#> += ;
+            that->nameTwo -= 0.001f;
+            }
+        std::cout << "U's nameTwo updated value: " << that->nameTwo << std::endl;
+        
+        return that->nameTwo * that->nameOne;
         }
```
Comment (matkatmusic) [inline main.cpp:137]: you've got some indentation issues here. 
- [ ] revise.

File: main.cpp [hunk 8]
```diff
@@ -1,133 +1,172 @@
...
+    if (smaller != nullptr)
+        std::cout << "\nthe smaller one is << " << smaller->name << std::endl; //9
+    
+    U nameThree;
     float updatedValue = 5.f;
-    std::cout << "[static func] <#name3#>'s multiplied values: " << <#structname2#>::<#staticFunctionA#>( , ) << std::endl;                  //11
+    std::cout << "[static func] nameThree's multiplied values: " << StructNameTwo::staticFunctionA(&nameThree, &updatedValue ) << std::endl;                  //11
+    
+    U nameFour;
+    // std::cout << "[member func] nameFour's multiplied values: " << nameFour.memberFunction( &updatedValue ) << std::endl;
```
Comment (matkatmusic) [inline main.cpp:160]: - [ ] uncomment this, make it work with your `U` member function `staticFunctionReplicate`

File: main.cpp [hunk 9]
```diff
@@ -1,133 +1,172 @@
...
     
-    U <#name4#>;
-    std::cout << "[member func] <#name4#>'s multiplied values: " << <#name4#>.<#memberFunction#>( &updatedValue ) << std::endl;
+    std::cout << "static function replicate in U: " << nameFour.staticFunctionReplicate(&updatedValue) << std::endl;
```
Comment (matkatmusic) [inline main.cpp:162]: - [ ] remove this

**Pull request merged**